labrad/servers/threaded_test_server.py

```diff
@@ -0,0 +1,156 @@
+"""
+### BEGIN NODE INFO
+[info]
+name = Python Threaded Test Server
+version = 1.0.0
+description = Basic python server.
+
+[startup]
+cmdline = %PYTHON% %FILE%
+timeout = 20
+
+[shutdown]
+message = 987654321
+timeout = 5
+### END NODE INFO
+"""
+
+import time
+
+from labrad import types as T, util
+from labrad.server import ThreadedServer, setting
+from labrad.units import m, s
+from labrad.util import hydrant
+
+
+class TestServer(ThreadedServer):
+    """Server to test labrad from python.
+
+    This server provides a number of settings that
+    are useful for testing labrad code, network profiling,
+    and testing the python labrad library.
+    """
+    name = 'Python Threaded Test Server'
+    testMode = True
+    shutdownMessage = 987654321
+
+    def initServer(self):
+        time.sleep(0.5)
+
+    def stopServer(self):
+        print 'before sleep'
+        time.sleep(0.5)
+        print 'after sleep'
+
+    def serverConnected(self, ID, name):
+        print 'server connected:', ID, name
+
+    def serverDisconnected(self, ID, name):
+        print 'server disconnected:', ID, name
+
+    def initContext(self, c):
+        c['delay'] = 1*s
+        c['dict'] = {}
+
+    @setting(2, "Delayed Echo", data='?')
+    def delayed_echo(self, c, data):
+        """Echo a packet after a specified delay."""
+        time.sleep(c['delay'][s])
+        return data
+
+    @setting(3, "Future Echo", data='?')
+    def future_echo(self, c, data):
+        """Echo a packet after a specified delay."""
+        f = self.client.manager.echo.future(data)
```

Comment:
It would be nice to have a method with a blocking client call as well as a future.